lib/cartopy/crs.py

```diff
@@ -459,6 +459,11 @@ def _rings_to_multi_polygon(self, rings, is_ccw):
                 if prep_polygon.contains(interior_ring):
                     holes.append(interior_ring)
                     interior_rings.remove(interior_ring)
+                elif polygon.crosses(interior_ring):
```

Comment:
`prep_polygon` doesn't have a `crosses()` method, but have you tried preparing the exterior ring instead? Something like:
```
prep_exterior = prep(polygon.exterior)
...
    elif prep_exterior.intersects(interior_ring)
```
NB. I've no idea if/when the prepared version might be worth using.